File: backend/entity/role.go
// entity/role.go
package entity

import "gorm.io/gorm"

type Role struct {
	gorm.Model
	RoleName    string `json:"role_name"`
	Description string `json:"description"`

	Users []User `gorm:"foreignKey:RoleID" json:"users,omitempty"`
}

File: frontend/src/interfaces/Role.ts
export interface Role {
  ID: number;
  role_name: string;
  description: string;
}

export interface CreateRoleRequest {
  role_name: string;
  description: string;
}

export interface UpdateRoleRequest {
  ID: number;
  role_name?: string;
  description?: string;
}
